File: 387. First Unique Character in a String/Python/Solution.py
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        counts = [0] * 26
        for item in s:
            counts[ord(item) - ord('a')] += 1
        for (index, item) in enumerate(s):
            if counts[ord(item) - ord('a')] == 1:
                return index
        return -1

File: 387. First Unique Character in a String/C++/Solution.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> counts(26, 0);
        for (auto item : s) {
            counts[item - 'a']++;
        }
        for (int i = 0; i < s.size(); i++) {
            if (counts[s[i] - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
};